### PR TITLE
Add package jblob-3.0

### PR DESCRIPTION
**Pull request by ajkotobi**
JBLOB-3.0: software for climate researchers and data management.

### CODE REVIEW

File: var/spack/repos/builtin/packages/jblob/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from spack import *
+import fileinput, os, shutil
+
+class Jblob(Package):
+    """The German Climate Computing Center (DKRZ: Deutsches Klimarechenzentrum GmbH) 
+       dsaprovides a Long Term Archiving Service for large research data sets which are relevant for climate or Earth system research.
+    """
+
+    homepage = "https://cera-www.dkrz.de/WDCC/ui/cerasearch"
+    url      = "http://cera-www.dkrz.de/jblob/jblob-3.0.zip"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jblob/package.py]:
```suggestion
    url      = "https://cera-www.dkrz.de/jblob/jblob-3.0.zip"
```
Does this still work?

File: var/spack/repos/builtin/packages/jblob/package.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    maintainers = ['ajkotobi']
+
+    version('3.0', sha256='576b5956358386a8832c6d1d13c410705e54888354a10cfd4f094513458067e4', expand=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jblob/package.py]:
```suggestion
    version('3.0', sha256='576b5956358386a8832c6d1d13c410705e54888354a10cfd4f094513458067e4')
```
That's the default

File: var/spack/repos/builtin/packages/jblob/package.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    depends_on('openjdk@1.8.0_202-b08:', type=('build', 'link', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jblob/package.py]: Does this work with other Java providers? In that case, I would do:
```suggestion
    depends_on('java@8:', type='run')
```
Java is only needed at run-time since you aren't building anything.

File: var/spack/repos/builtin/packages/jblob/package.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('JAVA_HOME', self.spec['java'].prefix)
+
+    def install(self, spec, prefix):
+
+        with open('jblob') as f:
+            replaced = f.read().replace('/opt/jblob-3.0', self.prefix)
+
+        with open('jblob', "w") as f:
+            f.write(replaced)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jblob/package.py]: It's more Spack-like to use:
```python
filter_file('/opt/jblob-' + self.version, prefix, 'jblob')
```

File: var/spack/repos/builtin/packages/jblob/package.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+        mkdir(prefix.bin)
+        copy('jblob', prefix.bin)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jblob/package.py]:
```suggestion
        install('jblob', prefix.bin)
```
The only difference should be permissions

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 6]
```diff
@@ -25,6 +25,7 @@ class Gromacs(CMakePackage):
     maintainers = ['junghans', 'marvinbernhardt']
 
     version('master', branch='master')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gromacs/package.py]: Was this change intentional or accidental?

File: var/spack/repos/builtin/packages/jblob/package.py (hunk 1 above)
Comment (ajkotobi) [reply var/spack/repos/builtin/packages/jblob/package.py]: Both http & https works but changed it to `https` link.

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 6 above)
Comment (ajkotobi) [reply var/spack/repos/builtin/packages/gromacs/package.py]: Those changes are accidental.

File: var/spack/repos/builtin/packages/jblob/package.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    homepage = "https://cera-www.dkrz.de/WDCC/ui/cerasearch"
+    url      = "https://cera-www.dkrz.de/jblob/jblob-3.0.zip"
+
+    maintainers = ['ajkotobi']
+
+    version('3.0', sha256='576b5956358386a8832c6d1d13c410705e54888354a10cfd4f094513458067e4')
+
+    depends_on('java@8:', type='run')
+
+    def setup_environment(self, spack_env, run_env):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jblob/package.py]: Use `setup_run_environment`, the previous API is deprecated

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 6 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gromacs/package.py]: Can you remove these changes from the PR?
